packages/fxa-settings/src/lib/integrations/integration-factory-flags.ts

```diff
@@ -0,0 +1,131 @@
+ *
+ * Note: this logic was ported from fxa-content-server app-start.js.
+ */
+export class DefaultIntegrationFactoryFlags implements IntegrationFlags {
+  protected get pathname() {
+    return this.urlQueryData.pathName;
+  }
+  constructor(private urlQueryData: UrlQueryData) {}
+
+  // TODO: Same method on the Relier flags, can we DRY?
+  isDevicePairingAsAuthority() {
+    return (
+      this._searchParam('redirect_uri') ===
+      Constants.DEVICE_PAIRING_AUTHORITY_REDIRECT_URI
+    );
+  }
+
+  isDevicePairingAsSupplicant() {
+    return DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP.test(this.pathname);
+  }
+
+  // TODO: Same method on the Relier flags, can we DRY?
+  isOAuth() {
+    return (
+      !!(
+        this._searchParam('client_id') ||
+        // verification
+        this._isOAuthVerificationSameBrowser()
+      ) ||
+      !!this._isOAuthVerificationDifferentBrowser() ||
+      // any URL with 'oauth' in the path.
+      /oauth/.test(this.pathname)
+    );
+  }
+
+  isOAuthWebChannel() {
+    // TODO
+    return false;
+  }
+
+  isChromeAndroid() {
+    // TODO
+    return false;
+  }
+
+  isSyncService() {
+    return this._isService(Constants.SYNC_SERVICE);
+  }
+
+  isV3DesktopContext() {
+    return this._searchParam('context') === Constants.FX_DESKTOP_V3_CONTEXT;
+  }
+
+  isOAuthSuccessFlow() {
+    const status = /oauth\/success/.test(this.pathname);
+
+    let clientId = '';
+    if (status) {
+      const pathname = this.pathname.split('/');
+      clientId = pathname[pathname.length - 1];
+    }
+
+    return {
+      status,
+      clientId,
+    };
+  }
+
+  isOAuthVerificationFlow(): boolean {
+    return !!this._searchParam('code');
+  }
+
+  private _isOAuthVerificationSameBrowser() {
+    return this._isVerification() && this._isService(this._getSavedClientId());
+  }
+
+  private _isVerification() {
+    return (
+      this._isSignUpVerification() ||
+      this._isPasswordResetVerification() ||
+      this._isReportSignIn()
+    );
+  }
+
+  private _isPasswordResetVerification() {
+    return this._searchParam('code') && this._searchParam('token');
+  }
+
+  private _isReportSignIn() {
+    return this.pathname === '/report_signin';
+  }
+
+  private _isSignUpVerification() {
+    return this._searchParam('code') && this._searchParam('uid');
+  }
+
+  private _searchParam(key: string) {
+    return this.urlQueryData.get(key);
+  }
+
+  private _isServiceSync() {
+    return this._isService(Constants.SYNC_SERVICE);
+  }
+
+  private _isService(compareToService: string) {
+    const service = this._searchParam('service');
+    return !!(service && compareToService && service === compareToService);
+  }
+
+  private _isOAuthVerificationDifferentBrowser() {
+    return this._isVerification() && this._isServiceOAuth();
+  }
+
+  private _isServiceOAuth() {
+    const service = this._searchParam('service');
+    return service && !this._isServiceSync();
```

Comment:
uber nit: not 100% sure what `const service = this._searchParam('service');` does.
Looking at `_isServiceSync()`, it calls `_isService(Constants.SYNC_SERVICE)`, and `_isService()` already seems to check `this._searchParam('service')` and then compare to the specified service.